### ISSUE TITLE
Avatar Privacy

### ISSUE THREAD

**Issue by gsantner** [vector-im/riot-android#2346]
Hi, as it seems like there is only support for one user picture (which can everybody see) - I thought to ask if that is gonna change at some point.

A user might want to use a personal picture that personal friends/family can only see, while public/all others see another - without face e.g..

Is there any support in Matrix for this or if not, is there any way this privacy option may get added? This is currently a public leaking bit of information.

**Comment by Matrixcoffee** [comment]
Matrix supports it, Riot does not.

I recommend opening an issue on Riot-Web as well, as that leads the design and the features. The mobile clients follow behind.

**Comment by gsantner** [comment]
I dont want to change room picture to be my picture. than when I open chat all I see is myself avatar, and not the one of the chat partner I do chat with (in 1:1 direct chats)

**Comment by Matrixcoffee** [comment]
I don't think you understand. Have a look at this: https://vault.webdevguru.co.uk/Matrix/per-room-nickname/

**Comment by gsantner** [comment]
No, this is about picture not realname. Too the main expectation is not room based, but global.

Real example from myself:
Have a matrix account with public known username, and association with my realname.
Now I want to chat with my father and others from family over Matrix. Non-Techies - this means they expect to see just my name and a real face icon.

For this father has a public unknown username, nothing that would identify a relation with myself, or give details about him.

In any situation its the goal that I can see his face picture (where no avatar at all should be public) and his real displayname. The same goes the other way round, and with other family members.

This is currently not possible, and thats why I reported this. What I call Privacy in most important level - to use Matrix without leaving public identifiable information.

Other protocols/messengers usually have an Option to set visibility for:
* Nobody/ Everybody
* Specific persons (users)
* Groups (Rooms)


And thats what I expect from Matrix protocol and clients to support too. Maybe not all cases, but the example with father has the main goals. This is nothing I see as optional feature, but a basic need.
I don't know what is currently possible or planned but I do hope, Matrix as speaking for open, decentralization and privacy does get an solution for this.

**Comment by tnyeanderson** [comment]
The per-room avatar and nickname are definitely critical features. Adding an option to edit your nickname per-room without the devtools would be an excellent and relatively simple start for this, then get started on the avatar options.
